source/include/internal/azure_iot_internal.h

```diff
@@ -0,0 +1,37 @@
+/* Copyright (c) Microsoft Corporation.
```

Comment:
How about putting this file in source folder, and do include using #include "azure_iot_internal.h" in c files. This way none of the build system has to change, be it IDE or cmake as current directory of source is  used for searching header file. 
This way we are telling customers to only include our inc path, while building our stuff. Other option do require them to include internal folder while building our stuff in IDE.

Reply:
If they include the `/source/include` folder this path still resolves since the include strategy should be 
```c
#include "internal/azure_iot_internal.h
```
The embedded SDK does it the same way. This way, if someone is going to include it, they have to explicitly put the `internal` prefix so they know this is an internal header.

Reply:
having internal in header file name should suffice the intention, I do not think you need folder name also to show that intention. Adding internal folder has downside on useability ... I think we should get embedded sdk to align to this.